File: apollo/src/main/java/com/lsxiao.apllo/annotations/Sticky.java
package com.lsxiao.apllo.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * write with Apollo
 * author:lsxiao
 * date:2017-04-22 02:20
 * github:https://github.com/lsxiao
 * zhihu:https://zhihu.com/people/lsxiao
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.CLASS)
public @interface Sticky {
}

File: core/src/main/java/com/lsxiao/apollo/core/annotations/Sticky.kt
package com.lsxiao.apollo.core.annotations

/**
 * write with Apollo
 * author:lsxiao
 * date:2017-04-22 02:20
 * github:https://github.com/lsxiao
 * zhihu:https://zhihu.com/people/lsxiao
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@kotlin.annotation.Retention
annotation class Sticky(val remove: Boolean = true)
